packages/manager/modules/telecom-dashboard/src/identity-check/form/telecom-dashboard-identity-check-form.controller.js

```diff
@@ -0,0 +1,87 @@
+export default class IdentityCheckFormCtrl {
+  /* @ngInject */
+  constructor(
+    coreConfig,
+    coreURLBuilder,
+    ovhPaymentMethodHelper,
+    IdentityCheckService,
+    TucToastError,
+  ) {
+    const { isValidIban, isValidBic } = ovhPaymentMethodHelper;
+
+    this.user = coreConfig.getUser();
+    this.isValidIban = isValidIban;
+    this.isValidBic = isValidBic;
+    this.IdentityCheckService = IdentityCheckService;
+    this.TucToastError = TucToastError;
+
+    this.isLoading = true;
+    this.isCreating = false;
+    this.isCancelling = false;
+    this.form = null;
+    this.procedure = null;
+    this.telecomLink = coreURLBuilder.buildURL('telecom', '#/');
+    this.model = {
+      bic: '',
+      iban: '',
+      ownerAddress: '',
+      ownerFirstName: '',
+      ownerLastName: '',
+    };
+  }
+
+  $onInit() {
+    const { name, firstname, address } = this.user;
+
+    this.model.ownerAddress = address;
+    this.model.ownerFirstName = firstname;
+    this.model.ownerLastName = name;
+
+    this.IdentityCheckService.getLastInProgressProcedure()
+      .then((procedure) => {
+        this.procedure = procedure;
+      })
+      .catch((error) => new this.TucToastError(error))
+      .finally(() => {
+        this.isLoading = false;
+      });
+  }
+
+  canCreateProcedure() {
+    return this.form?.$valid && !this.isCreating;
+  }
+
+  createProcedure() {
+    if (!this.canCreateProcedure()) return;
```

Comment:
This seems useless are you're disabling the button responsible for this action if `canCreateProcedure` is falsy

Reply:
Yes, I do know, but that was intended
- From the controller's perspective, it doesn't know about the view or how it uses its interface, since there are decoupled
- From unit tests perspective (yes, there are none but there should be), the `createProcedure` method should act in a stand alone way.
  - For that particular case, the procedure can't be created if the condition doesn't fit, regardless on how / when / where the method is called.
  - And for the case below, the line `const { id } = this.procedure;` could throw an error since `this.procedure` is initiated with a `null` value. Not to mention the following api call if `id` is not defined

Reply:
Done :+1: